src/main.rs

```diff
@@ -1084,20 +1113,42 @@ fn read_events(path: &Path, rdesc: &ReportDescriptor) -> Result<()> {
     let mut start_time: Option<Instant> = None;
     let mut last_timestamp: Option<Instant> = None;
     let mut data = [0; 1024];
+
+    let mut builder = libbpf_rs::RingBufferBuilder::new();
+    builder
+        .add(map_ringbuf, move |data| {
+            event_handler(data, &mut start_time)
```

Comment:
weird, I'm surprised this works. You're passing a mutable ref to `start_time` into the builder but later in the `f.read()` arms we change `start_time`. Really surprised rust doesn't throw the borrow checker at us here.

Reply:
Not sure how this is possible, but it turns out that the value we get here is not shared with the main thread. So if I update in hidraw event, the value is still None in the bpf handler. And the other way around.

I might as well get a different value.